src/components/RegisterModal/RegisterModal.js

```diff
@@ -0,0 +1,247 @@
+import { Link, useHistory } from "react-router-dom/cjs/react-router-dom";
+import ModalWithForm from "../ModalWithForm/ModalWithForm";
+import { useForm } from "../../hooks/useForm";
+import { useFormAndValidation } from "../../hooks/useFormAndValidation";
+import { useState } from "react";
+import TinyPopup from "../TinyPopUp/TinyPopUp";
+
+const RegisterModal = ({
+  handleCloseModal,
+  isOpen,
+  onSubmit,
+  buttonText,
+  openLogInModal,
+}) => {
+  const [isTooltipDOBVisible, setIsTooltipDOBVisible] = useState(false);
+  const [isTooltipBirthPlaceVisible, setIsTooltipBirthPlaceVisible] =
+    useState(false);
+  const [isTooltipMaritalVisible, setIsTooltipMaritalVisible] = useState(false);
+  const [isTooltipSexOrientationVisible, setIsTooltipSexOrientationVisible] =
+    useState(false);
+
+  const { values, handleChange, errors, isValid } = useFormAndValidation({
+    name: "",
+    dob: "",
+    placeOfBirth: "",
+    maritalStatus: "",
+    sexualOrientation: "",
+    email: "",
+    password: "",
+  });
+
+  // const history = useHistory();
+
+  const handleFormSubmitRegister = (e) => {
+    e.preventDefault();
+    if (isValid) {
+      onSubmit(values);
+    }
+  };
+
+  const handleOpenLogin = (e) => {
+    e.preventDefault();
+    openLogInModal();
+  };
+
+  return (
+    <div className="register">
+      <ModalWithForm
+        title="Sign Up"
```

Comment:
* Please, pass `isValid` into the props of `ModalWithForm` to disable the submit button if the form is **not** valid